context.go

```diff
@@ -409,40 +410,41 @@ func (m *Macaron) SetDefaultCookieSecret(secret string) {
 
 // SetSecureCookie sets given cookie value to response header with default secret string.
 func (ctx *Context) SetSecureCookie(name, value string, others ...interface{}) {
-	ctx.SetSuperSecureCookie(defaultCookieSecret, name, value, others...)
+	ctx.SetSuperSecureCookie([]byte(defaultCookieSecret), name, value, others...)
 }
 
 // GetSecureCookie returns given cookie value from request header with default secret string.
 func (ctx *Context) GetSecureCookie(key string) (string, bool) {
-	return ctx.GetSuperSecureCookie(defaultCookieSecret, key)
+	return ctx.GetSuperSecureCookie([]byte(defaultCookieSecret), key)
 }
 
 // SetSuperSecureCookie sets given cookie value to response header with secret string.
-func (ctx *Context) SetSuperSecureCookie(secret, name, value string, others ...interface{}) {
-	m := md5.Sum([]byte(secret))
-	secret = hex.EncodeToString(m[:])
-	text, err := com.AESEncrypt([]byte(secret), []byte(value))
+func (ctx *Context) SetSuperSecureCookie(password []byte, name, value string, others ...interface{}) []byte {
```

Comment:
The thing want to be encrypted isn't necessarily a password, but something want to protect and hard to guess.

>Would you rather I keep password as a string and convert it to []byte later on when it's used? I just thought we could avoid the conversion if it was passed in directly as a []byte.

But you're actually making callers to convert, in the web context, mostly likely call will hold a string.